app/controllers/users_controller.rb

```diff
@@ -0,0 +1,48 @@
+class UsersController < ApplicationController
+  get '/signup' do
+    if logged_in?
+      redirect to "/tweets"
+    else
+      erb :'/users/signup'
+    end
+  end
+
+  post '/signup' do
+    @user = User.new(params)
+    if @user.username.empty? || @user.email.empty? || @user.password_digest == nil
```

Comment:
I think you could use [ActiveRecord validations](http://guides.rubyonrails.org/active_record_validations.html) (e.g. `validates_presence_of :username`) to perform these types of checks more easily.